### BC Hydro IPP - Resource Analysis
#### Wind

BC Hydro acquires power from Independent Power Producers (IPPs) to help meet B.C.'s electricity needs.
IPPs include power production companies, municipalities, First Nations and customers.

The analysis is carried out on the most recent [IPP supply list](https://www.bchydro.com/content/dam/BCHydro/customer-portal/documents/corporate/independent-power-producers-calls-for-power/independent-power-producers/ipp-supply-list-in-operation.pdf) and [IPP supply map](https://www.bchydro.com/content/dam/BCHydro/customer-portal/documents/corporate/independent-power-producers-calls-for-power/independent-power-producers/ipp-supply-map.pdf) obtained from [BC Hydro Website](https://www.bchydro.com/work-with-us/selling-clean-energy/meeting-energy-needs/how-power-is-acquired.html).

In [1]:
import pandas as pd
import altair as alt

In [2]:
data = pd.read_csv('IPP__REGION_list.csv')

#### Initial Description of Data


As of April 1, 2025, BC Hydro has 119 Electricity Purchase Agreements (EPAs) with IPPs whose projects achieved commercial operation, as defined in their EPA and are considered to be in operation. In total, these projects are capable of delivering approximately 18,662 gigawatt hours of annual supply and approximately 5,298 megawatts of capacity.

In [3]:
len(data)

119

In [4]:
IPP_Capacity = round(data['Capacity(MW)'].sum(),2)
IPP_Capacity

5298.84

In [5]:
IPP_Energy = round(data['Energy(GWh/yr)'].sum(),2)
IPP_Energy

18662.3

In [6]:
data.replace({'Sub_region':{'CI':'Central Interior',
                               'LM':'Lower Mainland',
                               'VI':'Vancouver Island',
                               'NC':'North Coast',
                               'EK':'East Kootenay',
                               'PR':'Peace River',
                               'KLN':'Kelly Lake Nicola',
                               'RSAC':'Revelstoke Ashton Creek',
                               'SK':'Selkirk'}}, inplace = True)

In [7]:
peace_list = ['Central Interior','Peace River']

In [8]:
columbia_list = ['Revelstoke Ashton Creek', 'Kelly Lake Nicola', 'Mica', 'Selkirk', 'East Kootenay']

In [9]:
data.loc[data['Sub_region'].isin(columbia_list),'Region'] = 'Columbia Region'
data.loc[data['Sub_region'].isin(peace_list),'Region'] = 'Peace Region'
data.loc[data['Sub_region'] == 'Lower Mainland','Region'] = 'Lower Mainland'
data.loc[data['Sub_region'] == 'North Coast','Region'] = 'North Coast'
data.loc[data['Sub_region'] == 'Vancouver Island','Region'] = 'Vancouver Island'

In [10]:
columns = data.columns.to_list()
for column in columns:
    if data[column].dtype == 'object':
        data[column] = data[column].str.strip()

The capacity and GWhr contribution of various resources by Type is given by the data_capacity_type and data_energy_type respectively 

In [11]:
data_capacity_type = data.groupby('Type').agg({'Capacity(MW)':'sum'}).sort_values(by = 'Capacity(MW)',ascending = False).reset_index()
data_energy_type = data.groupby('Type').agg({'Energy(GWh/yr)':'sum'}).sort_values(by = 'Energy(GWh/yr)',ascending = False).reset_index()

In [12]:
data_energy_type

Type  Energy(GWh/yr)
0  Non-Storage Hydro          7336.4
1      Storage Hydro          4933.3
2            Biomass          2977.6
3               Wind          2161.8
4  Gas Fired Thermal           904.7
5                MSW           166.0
6                ERG           141.0
7             Biogas            37.6
8              Solar             3.9

In [13]:
data_capacity_type

Type  Capacity(MW)
0  Non-Storage Hydro      1983.200
1      Storage Hydro      1292.100
2            Biomass       817.900
3               Wind       747.000
4  Gas Fired Thermal       403.800
5                MSW        24.800
6                ERG        22.800
7             Biogas         5.155
8              Solar         2.090

The type of resource contributing to the largest capacity and energy is Wind 

#### Wind 

The kinetic energy in wind can be converted into useful forms of energy such as mechanical energy or electricity. Wind energy has been harnessed for centuries to propel sailing vessels and to turn grist mills and water pumps.

Today, wind is used increasingly to generate electricity. Turbines with large propellers are erected on wind farms located in strategic areas that have good wind regimes and that are in proximity to existing electrical grids. Wind energy is captured only when the wind speed is sufficient to move the turbine blades, but not in high winds when the turbine might be damaged if operated.([source](https://natural-resources.canada.ca/energy-sources/renewable-energy/wind-energy))

BC Hydro has Electricity Purchase Agreement for Wind with 10 Independent Power Producers. In total, these 10 projects are capable of delivering approximately 2161.8 gigawatt hours of annual supply and approximately 747 megawatts of capacity. The capacity and GWhr contribution of Wind in various regions are given by the dataframes energy_W, capacity_W respectively. 

In [47]:
data_W = data[data['Type'] == 'Wind'].reset_index(drop=True)
energy_W = data_W.groupby('Region').agg({'Energy(GWh/yr)':'sum',
                                         'Capacity(MW)':'sum'}).sort_values(by = 'Energy(GWh/yr)',ascending = False).reset_index()
capacity_W = data_W.groupby('Region').agg({'Capacity(MW)':'sum'}).sort_values(by = 'Capacity(MW)',ascending = False).reset_index()

In [48]:
data_W

Sub_region                   Project_name  \
0        Peace River                     Dokie Wind   
1        Peace River    Sukunka Wind Energy Project   
2        Peace River  Zonnebeke Wind Energy Project   
3   Vancouver Island        Bear Mountain Wind Park   
4   Vancouver Island                Cape Scott Wind   
5  Kelly Lake Nicola                  Shinish Creek   
6        Peace River                    Meikle Wind   
7        Peace River                   Quality Wind   
8        Peace River        Moose Lake Wind Project   
9  Kelly Lake Nicola              Pennask Wind Farm   

                                          IPP_seller       Location  Type  \
0                          Dokie General Partnership       Chetwynd  Wind   
1           Sukunka Wind Project Limited Partnership       Chetwynd  Wind   
2         Zonnebeke Wind Project Limited Partnership       Chetwynd  Wind   
3             Bear Mountain Wind Limited Partnership   Dawson Creek  Wind   
4                                 Cape Scott Wind LP     Port Hardy  Wind   
5  Wind Farm Zero Emissions Shinish Creek Limited...     Summerland  Wind   
6             Meikle Wind Energy Limited Partnership  Tumbler Ridge  Wind   
7                                 Capital Power L.P.  Tumbler Ridge  Wind   
8                Moose Lake Wind Limited Partnership  Tumbler Ridge  Wind   
9         Zero Emissions Pennask Limited Partnership       Westbank  Wind   

   Year            Call_process  Capacity(MW)  Energy(GWh/yr)  \
0  2009          Negotiated EPA         144.0           375.0   
1  2010  Standing Offer Program          15.0            49.7   
2  2010  Standing Offer Program          15.0            45.6   
3  2006               Open Call         102.0           197.0   
4  2010        Clean Power Call          99.2           316.3   
5  2010  Standing Offer Program          15.0            54.5   
6  2010        Clean Power Call         184.6           541.0   
7  2010        Clean Power Call         142.2           476.9   
8  2010  Standing Offer Program          15.0            56.0   
9  2010  Standing Offer Program          15.0            49.8   

             Region  
0      Peace Region  
1      Peace Region  
2      Peace Region  
3  Vancouver Island  
4  Vancouver Island  
5   Columbia Region  
6      Peace Region  
7      Peace Region  
8      Peace Region  
9   Columbia Region

In [49]:
energy_W

Region  Energy(GWh/yr)  Capacity(MW)
0      Peace Region          1544.2         515.8
1  Vancouver Island           513.3         201.2
2   Columbia Region           104.3          30.0

In [44]:
energy_W['Energy(GWh/yr)'].sum()

2161.8

In [45]:
capacity_W

Region  Capacity(MW)
0      Peace Region         515.8
1  Vancouver Island         201.2
2   Columbia Region          30.0

In [38]:
capacity_W['Capacity(MW)'].sum()

747.0

For Wind, the capacity and energy contributions of each region individually analysed 

#### 1. Lower Mainland


For the lower mainland region, there are no wind energy IPP Project in operation. 


In [18]:
data_LM = data[(data['Type'] == 'Wind') & (data['Region']=='Lower Mainland')].reset_index(drop=True)

In [19]:
len(data_LM)

0

#### 2. Vancouver Island


For the Vancouver Island region, the capacity and GWhr contribution of the Wind resource by location is given in the capacity_VI_Loc and energy_VI_Loc dataframes respectively.

In [50]:
data_VI = data[(data['Type'] == 'Wind') & (data['Region']=='Vancouver Island')].reset_index(drop=True)
energy_VI_Loc = data_VI.groupby('Location').agg({'Energy(GWh/yr)':'sum',
                                                'Capacity(MW)':'sum'}).sort_values(by = 'Energy(GWh/yr)',ascending = False).reset_index()
capacity_VI_Loc = data_VI.groupby('Location').agg({'Capacity(MW)':'sum'}).sort_values(by = 'Capacity(MW)',ascending = False).reset_index()

In [51]:
energy_VI_Loc

Location  Energy(GWh/yr)  Capacity(MW)
0    Port Hardy           316.3          99.2
1  Dawson Creek           197.0         102.0

In [22]:
capacity_VI_Loc

Location  Capacity(MW)
0  Dawson Creek         102.0
1    Port Hardy          99.2

The bar chart energy_VI_Loc_chart, shows the energy contribution for each of these Wind projects and the call process by which it was acquired. 

The bar chart capacity_VI_Loc_chart, shows the capacity contribution for each of these Wind projects and the call process by which it was acquired.


In [23]:
title = alt.TitleParams('The Annual GWhr contribution of Wind in Vancouver Island by IPP Projects')

capacity_IPP = alt.Chart(data_VI, title = title).mark_bar().encode(
                    alt.Y('Project_name', title='Project Name',sort='x'),
                    alt.X('sum(Energy(GWh/yr))', title='Energy(GWh/yr)'),
                    alt.Color('Call_process', title = 'Call Process')).properties( width=300)
energy_VI_Loc_chart = capacity_IPP + capacity_IPP.mark_text(
    color = 'black', align='left', dx = 3).encode(text='sum(Energy(GWh/yr))',color=alt.value('black'))
energy_VI_Loc_chart

alt.LayerChart(...)

In [24]:
title = alt.TitleParams('The Capacity contribution of Wind in Vancouver Island by IPP Projects')

capacity_IPP = alt.Chart(data_VI, title = title).mark_bar().encode(
                    alt.Y('Project_name', title='Project Name',sort='x'),
                    alt.X('sum(Capacity(MW))', title='Capacity(MW)'),
                    alt.Color('Call_process', title = 'Call Process')).properties( width=300)
cap_VI_Loc_chart = capacity_IPP + capacity_IPP.mark_text(
    color = 'black', align='left', dx = 3).encode(text='sum(Capacity(MW))',color=alt.value('black'))
cap_VI_Loc_chart

alt.LayerChart(...)

#### 3. Peace Region

For Peace region, the capacity and GWhr contribution of the Wind resource by location is given in the capacity_PR_Loc and energy_PR_Loc dataframes respectively.

In [52]:
data_PR = data[(data['Type'] == 'Wind') & (data['Region']=='Peace Region')].reset_index(drop=True)
energy_PR_Loc = data_PR.groupby('Location').agg({'Energy(GWh/yr)':'sum',
                                                'Capacity(MW)':'sum'}).sort_values(by = 'Energy(GWh/yr)',ascending = False).reset_index()
capacity_PR_Loc = data_PR.groupby('Location').agg({'Capacity(MW)':'sum'}).sort_values(by = 'Capacity(MW)',ascending = False).reset_index()

In [53]:
energy_PR_Loc

Location  Energy(GWh/yr)  Capacity(MW)
0  Tumbler Ridge          1073.9         341.8
1       Chetwynd           470.3         174.0

In [27]:
capacity_PR_Loc

Location  Capacity(MW)
0  Tumbler Ridge         341.8
1       Chetwynd         174.0

The bar chart energy_PR_Loc_chart, shows the energy contribution of the project and the call process by which it was acquired. 

The bar chart capacity_PR_Loc_chart, shows the capacity contribution of each of these projects and the call process by which it was acquired.

In [28]:
title = alt.TitleParams('The Annual GWhr contribution of Wind in Peace Region by IPP Project')

capacity_IPP = alt.Chart(data_PR, title = title).mark_bar().encode(
                    alt.Y('Project_name', title='Project Name',sort='x'),
                    alt.X('sum(Energy(GWh/yr))', title='Energy(GWh/yr)'),
                    alt.Color('Call_process', title = 'Call Process')).properties( width=300)
energy_PR_Loc_chart = capacity_IPP + capacity_IPP.mark_text(
    color = 'black', align='left', dx = 3).encode(text='sum(Energy(GWh/yr))',color=alt.value('black'))
energy_PR_Loc_chart

alt.LayerChart(...)

In [29]:
title = alt.TitleParams('The Capacity contribution of Wind in Peace Region by IPP Project')

capacity_IPP = alt.Chart(data_PR, title = title).mark_bar().encode(
                    alt.Y('Project_name', title='Project Name',sort='x'),
                    alt.X('sum(Capacity(MW))', title='Capacity(MW)'),
                    alt.Color('Call_process', title = 'Call Process')).properties( width=300)
cap_PR_Loc_chart = capacity_IPP + capacity_IPP.mark_text(
    color = 'black', align='left', dx = 3).encode(text='sum(Capacity(MW))',color=alt.value('black'))
cap_PR_Loc_chart

alt.LayerChart(...)

#### 4. Columbia Region

For Columbia region, the capacity and GWhr contribution of the Wind resource by location is given in the capacity_CR_Loc and energy_CR_Loc dataframes respectively.

In [30]:
data_CR = data[(data['Type'] == 'Wind') & (data['Region']=='Columbia Region')].reset_index(drop=True)
energy_CR_Loc = data_CR.groupby('Location').agg({'Energy(GWh/yr)':'sum'}).sort_values(by = 'Energy(GWh/yr)',ascending = False).reset_index()
capacity_CR_Loc = data_CR.groupby('Location').agg({'Capacity(MW)':'sum'}).sort_values(by = 'Capacity(MW)',ascending = False).reset_index()

In [31]:
energy_CR_Loc

Location  Energy(GWh/yr)
0  Summerland            54.5
1    Westbank            49.8

In [32]:
capacity_CR_Loc

Location  Capacity(MW)
0  Summerland          15.0
1    Westbank          15.0

The bar chart energy_CR_Loc_chart, shows the energy contribution of the project and the call process by which it was acquired. 

The bar chart capacity_CR_Loc_chart, shows the capacity contribution of each of these projects and the call process by which it was acquired.

In [33]:
title = alt.TitleParams('The Annual GWhr contribution of Wind in Columbia Region by IPP Project')

capacity_IPP = alt.Chart(data_CR, title = title).mark_bar().encode(
                    alt.Y('Project_name', title='Project Name',sort='x'),
                    alt.X('sum(Energy(GWh/yr))', title='Energy(GWh/yr)'),
                    alt.Color('Call_process', title = 'Call Process')).properties( width=300)
energy_CR_Loc_chart = capacity_IPP + capacity_IPP.mark_text(
    color = 'black', align='left', dx = 3).encode(text='sum(Energy(GWh/yr))',color=alt.value('black'))
energy_CR_Loc_chart

alt.LayerChart(...)

In [34]:
title = alt.TitleParams('The Capacity contribution of Wind in Columbia Region by IPP Project')

capacity_IPP = alt.Chart(data_CR, title = title).mark_bar().encode(
                    alt.Y('Project_name', title='Project Name',sort='x'),
                    alt.X('sum(Capacity(MW))', title='Capacity(MW)'),
                    alt.Color('Call_process', title = 'Call Process')).properties( width=300)
cap_CR_Loc_chart = capacity_IPP + capacity_IPP.mark_text(
    color = 'black', align='left', dx = 3).encode(text='sum(Capacity(MW))',color=alt.value('black'))
cap_CR_Loc_chart

alt.LayerChart(...)

#### 5. North Coast

For North Coast region, there are no wind energy IPP Project in operation. 


In [35]:
data_NC = data[(data['Type'] == 'Wind') & (data['Region']=='North Coast')].reset_index(drop=True)

In [36]:
len(data_NC)

0

#### Conclusion

This study was cariied out to learn about the capacity and energy distribution of Wind IPP resources around various regions in the province. 